### PR TITLE
[AIRFLOW-5049] Add validation for src_fmt_configs in bigquery hook

### PR DESCRIPTION
**Pull request by DPGrev**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira]( https://issues.apache.org/jira/browse/AIRFLOW-5049) issues and references them in the PR title. 

https://issues.apache.org/jira/browse/AIRFLOW-5049

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

Wrong entries for the `src_fmt_configs` in the bigquery hook are silently ignored. This PR adds validation of these entries and throws an error when incorrect.

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

A test was added for `_validate_src_fmt_configs`

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

File: airflow/contrib/hooks/bigquery_hook.py [hunk 1]
```diff
@@ -1199,11 +1185,26 @@ def run_load(self,  # pylint: disable=too-many-locals,too-many-arguments,invalid
...
             'AVRO': ['useAvroLogicalTypes'],
         }
+
         valid_configs = src_fmt_to_configs_mapping[source_format]
-        src_fmt_configs = {
-            k: v
-            for k, v in src_fmt_configs.items() if k in valid_configs
-        }
+
+        # if following fields are not specified in src_fmt_configs,
```
Comment (potiuk) [inline airflow/contrib/hooks/bigquery_hook.py:1253]: That screams for extracting to common method reused in two pleaces.

**Review by potiuk — CHANGES_REQUESTED** [review]

**Comment by mik-laj** [comment]
Are you planning to continue working on this change? This week, I would like to deal with PR reviews related to GCP. I would be happy if you would respond to all comments

**Comment by DPGrev** [comment]
I will be picking this up later this week.

File: airflow/contrib/hooks/bigquery_hook.py (hunk 1 above)
Comment (DPGrev) [reply airflow/contrib/hooks/bigquery_hook.py:1253]: Sure I will make the change!

Comment (DPGrev) [reply airflow/contrib/hooks/bigquery_hook.py:1253]: @potiuk Pushed the change

**Review by BasPH — APPROVED** [review]

**Pull request merged**